pkg/kubectl/cmd/delete.go

```diff
@@ -138,10 +142,18 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		}
 	}
 
+	gracePeriod := cmdutil.GetFlagInt(cmd, "grace-period")
+	if cmdutil.GetFlagBool(cmd, "now") {
+		if gracePeriod != -1 {
+			return fmt.Errorf("--now and --grace-period cannot be specified together")
```

Comment:
we can allow `--now` and `--grace-period=0` to be set together (although it's redundant)